src/login/connect_server.h

```diff
@@ -0,0 +1,2257 @@
+                            ref<uint16>(CharList, charListOffset + 58) = static_cast<uint16>(sql->GetUIntData(7));  // head
+                            ref<uint16>(CharList, charListOffset + 60) = static_cast<uint16>(sql->GetUIntData(8));  // body
+                            ref<uint16>(CharList, charListOffset + 62) = static_cast<uint16>(sql->GetUIntData(9));  // hands
+                            ref<uint16>(CharList, charListOffset + 64) = static_cast<uint16>(sql->GetUIntData(10)); // legs
+                            ref<uint16>(CharList, charListOffset + 66) = static_cast<uint16>(sql->GetUIntData(11)); // feet
+                            ref<uint16>(CharList, charListOffset + 68) = static_cast<uint16>(sql->GetUIntData(12)); // main
+                            ref<uint16>(CharList, charListOffset + 70) = static_cast<uint16>(sql->GetUIntData(13)); // sub
+
+                            ref<uint8>(CharList, charListOffset + 72)  = static_cast<uint8>(zone);
+                            ref<uint16>(CharList, charListOffset + 78) = zone;
+
+                            ++i;
+                        }
+                    }
+
+                    // the filtering above removes any non-GM characters so
+                    // at this point we need to make sure stop players with empty lists
+                    // from logging in or creating new characters
+                    if (maintMode > 0 && i == 0)
+                    {
+                        if (auto data = session.view_session.get())
+                        {
+                            generateErrorMessage(data->data_, 321);
+                            data->do_write(0x24);
+                        }
+                        ShowWarning(fmt::format("char:({}) attmpted login during maintenance mode (0xA2). Sending error to client.", session.accountID));
+                        return;
+                    }
+
+                    if (auto data = session.data_session.get())
+                    {
+                        uList[1] = 0x10;
+                        std::memset(data->data_, 0, sizeof(data_));
+                        std::memcpy(data->data_, uList, 0x148);
+                        data->do_write(0x148);
+                    }
+
+                    if (auto data = session.view_session.get())
+                    {
+                        unsigned char hash[16];
+                        md5((unsigned char*)(CharList), hash, 2272);
+                        std::memcpy(CharList + 12, hash, 16);
+
+                        std::memset(data->data_, 0, sizeof(data_));
+                        std::memcpy(data->data_, CharList, 2272);
+                        data->do_write(2272);
+                    }
+                }
+            }
+            break;
+            case 0xA2: // 162 : "Notifying lobby of current selections" pt 2
+            {
+                LOBBY_A2_RESERVEPACKET(ReservePacket);
```

Comment:
All of these macro things: LOBBY_A2_RESERVEPACKET
Can we refactor them out to be functions that return an object instead of dumping something into the current scope?

Reply:
Refactored out into oblivion, now is a proper class provided by atom0s